superset-frontend/src/views/CRUD/hooks.ts

```diff
@@ -299,64 +301,85 @@ export function useSingleViewResource<D extends object = any>(
   };
 }
 
-// the hooks api has some known limitations around stale state in closures.
-// See https://github.com/reactjs/rfcs/blob/master/text/0068-react-hooks.md#drawbacks
-// the useRef hook is a way of getting around these limitations by having a consistent ref
-// that points to the most recent value.
+enum FavStarClassName {
+  CHART = 'slice',
+  DASHBOARD = 'Dashboard',
+}
+
+type FavoriteStatusResponse = {
+  result: Array<{
+    id: string;
+    value: boolean;
+  }>;
+};
+
+const favoriteApis = {
+  chart: makeApi<string, FavoriteStatusResponse>({
```

Comment:
You can also pass a json object to the request payload, which would be better suited for large search params and strong typing.

In the future, we may even add built-in rison support so that rison requests can have typing, too.

Reply:
Yup, just a simple `encodeRison` flag would be enough to pass a properly typed object yet still have it issue a rison request.

Reply:
I'd vote for `requestType: "rison"` so to not add a new config option.